src/backend/core/api/viewsets.py

```diff
@@ -1237,6 +1239,70 @@ def ai_translate(self, request, *args, **kwargs):
 
         return drf.response.Response(response, status=drf.status.HTTP_200_OK)
 
+    @drf.decorators.action(
+        detail=True,
+        methods=["get"],
+        name="",
+        url_path="cors-proxy",
+    )
+    def cors_proxy(self, request, *args, **kwargs):
+        """
+        GET /api/v1.0/documents/<resource_id>/cors-proxy
+        Act like a proxy to fetch external resources and bypass CORS restrictions.
+        """
+        doc = self.get_object()
+
+        url = request.query_params.get("url")
+        if not url:
+            return drf.response.Response(
+                {"detail": "Missing 'url' query parameter"},
+                status=drf.status.HTTP_400_BAD_REQUEST,
+            )
+
+        if not doc.content:
+            return drf.response.Response(
+                {"detail": "Document content is empty"},
+                status=drf.status.HTTP_400_BAD_REQUEST,
+            )
+
+        url = unquote(url)
+
+        decoded_bytes = base64.b64decode(doc.content)
+        if not url.encode("utf-8") in decoded_bytes:
+            return drf.response.Response(
+                {"detail": "URL not found in document content"},
+                status=drf.status.HTTP_404_NOT_FOUND,
+            )
```

Comment:
This part actually can create problems.
3 kind of problems:
- the front save the doc every minutes (if a diff), if the image is set and does not have time to be saved and the user want to export, the endpoint will fails
- we are able to add template, the template is not part of the document content, we can have images in the template
- soon we will have the header / footer feature, same problem, they are not part of the `doc.content` but can have images.